### ISSUE TITLE
Implement wlr-foreign-toplevel request_minimize event

### ISSUE THREAD

**Issue by LBCrion** [swaywm/sway#6457]
i3 does not have a concept of a minimized window, but users commonly treat scratchpad as a viable alternative to minimization. To keep with i3 standards, can we implement wlr-foreign-toplevel minimize request as move to and from scratchpad? This approach keeps with i3 functionality and provides a more complete implementation of the foreign toplevel protocol.

**Comment by LBCrion** [comment]
A simple patch vs 1.6.1 to implement this functionality:
[sway-wlr-foreign-toplevel-minimize.txt](https://github.com/swaywm/sway/files/7026956/sway-wlr-foreign-toplevel-minimize.txt)

**Comment by emersion** [comment]
We don't do anything on xdg-shell `request_minimize`. Marking a window as minimized through wlr-foreign-toplevel would make things inconsistent.

**Comment by LBCrion** [comment]
I'm not suggesting marking the window as minimized. Just moving it to scratchpad. The implementation just provides applications (taskbars most likely) with a wayland interface to move windows to and from scratchpad.

**Comment by emersion** [comment]
i3 doesn't do it, so I don't think we should.

**Comment by emersion** [comment]
For reference, i3 discussion about this: https://github.com/i3/i3/pull/1457

**Comment by emersion** [comment]
The crash can be fixed. It's orthogonal to the proposal described in this issue.

**Comment by Iss-in** [comment]
> I would also ask what the purpose of the scratchpad is if not to act as a minimize?

yes exact;y, its an awesome functionality that is hindered by treating it as something that is not minimized.  usecase involving swiftly hiding/unhiding background floating apps like pomodoro and sticky notes via taskbar

@emersion  pls look into it, it really brings a lot of convenience here

**Comment by WhyNotHugo** [comment]
> I would also ask what the purpose of the scratchpad is if not to act as a minimize?

I mounted something via `sshfs` on a shell, and want the terminal window out of the way forever -> stratchpad.

The scratchpad is not really an equivalent to minimisation; windows sent to the scratchpad lose their workspace too. Usually, minimised windows remain on that workspace and are not accessible on other workspaces. The semantics don't really match.

**Comment by tmccombs** [comment]
What if it doesn't move to scratchpad by default, but there is a configuration option to allow that behavior for people who want it.

**Comment by breno-ca** [comment]
I’m glad to know I’m not the only one missing this 😂.

I’ve set up specific shortcuts for toggling some apps from my scratchpad on the screen. I don’t use workspaces because I usually organize everything across my two monitors, configuring floating windows in specific positions.

However, some of these apps still have that darn minimize icon in their embedded top bar. After 18 years of using a computer and getting used to minimizing apps through that icon, it’s really hard not to click it out of habit, expecting it to send the app to the scratchpad.

It would be really interesting if this were a configurable option in Sway.

**Comment by WhyNotHugo** [comment]
> However, some of these apps still have that darn minimize icon in their embedded top bar. After 18 years of using a computer and getting used to minimizing apps through that icon, it’s really hard not to click it out of habit, expecting it to send the app to the scratchpad.

Tangential: This sounds like a bug in the application; sway properly reports that it doesn't support minimisation, so the application should not show an icon for an unsupported feature.
